L3/CustomViews/TextFields/CardTextField.swift

```diff
@@ -0,0 +1,278 @@
+        }
+    }
+    
+    func onChangeComponentColour(for item: TextFieldComponents, with color: UIColor) {
+        switch item {
+        case .border:
+            textFieldBorderColor = color
+            defaultBorderColor = color
+        case .counter:
+            counter.textColor = color
+            defaultCounterColor = color
+        case .underline:
+            bottomBorder.backgroundColor = color
+            defaultBorderColor = color
+        case .switcher:
+            switcher.onTintColor = color
+            defaultSwitcherColor = color
+        case .placehonder:
+            let attributedPlaceholder = NSAttributedString(string: Constants.textFieldPlaceholder, attributes: [NSAttributedString.Key.foregroundColor: color])
+            textField.attributedPlaceholder = attributedPlaceholder
+            defaultPlacehonderColor = color
+        case .title:
+            titleLabel.textColor = color
+            defaultTitleColor = color
+        case .text:
+            textField.textColor = color
+            defaultTextColor = color
+        }
+    }
+    
+    private func configureLayout() {
+        counter.translatesAutoresizingMaskIntoConstraints = false
+        errorMessage.translatesAutoresizingMaskIntoConstraints = false
+        titleLabel.translatesAutoresizingMaskIntoConstraints = false
+        bottomBorder.translatesAutoresizingMaskIntoConstraints = false
+        
+        bottomBorderHeightConstraintLight = bottomBorder.heightAnchor.constraint(equalToConstant: 1)
+        bottomBorderHeightConstraintBold = bottomBorder.heightAnchor.constraint(equalToConstant: 2)
+        bottomBorderHeightConstraintLight?.isActive = true
+        
+        errorMessageHeightAnchor = errorMessage.heightAnchor.constraint(equalToConstant: 0)
+        errorMessageHeightAnchor?.isActive = true
+        
+        NSLayoutConstraint.activate([
+            textField.topAnchor.constraint(equalTo: topAnchor, constant: 15),
+            textField.leadingAnchor.constraint(equalTo: leadingAnchor, constant: Const.padding),
+            textField.trailingAnchor.constraint(equalTo: trailingAnchor, constant: -Const.padding),
+            textField.heightAnchor.constraint(equalToConstant: 24),
+            
+            bottomBorder.leadingAnchor.constraint(equalTo: textField.leadingAnchor, constant: switcher.frame.width + 12),
+            bottomBorder.trailingAnchor.constraint(equalTo: textField.trailingAnchor),
+            bottomBorder.topAnchor.constraint(equalTo: textField.bottomAnchor, constant: 2),
+            
+            counter.topAnchor.constraint(equalTo: textField.bottomAnchor, constant: 5),
+            counter.trailingAnchor.constraint(equalTo: textField.trailingAnchor),
+            
+            errorMessage.topAnchor.constraint(equalTo: counter.bottomAnchor),
+            errorMessage.bottomAnchor.constraint(equalTo: bottomAnchor, constant: -Const.padding),
```

Comment:
I see you are adding always same padding, so may it would be better to use 

`static let layoutMargins: NSDirectionalEdgeInsets = .init(top: 0, leading: 20, bottom: 0, trailing: 20)`